apps/dashboard/src/components/pay/PayAnalytics/hooks/usePayCustomers.ts

```diff
@@ -0,0 +1,78 @@
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { useLoggedInUser } from "../../../../@3rdweb-sdk/react/hooks/useLoggedInUser";
+import { THIRDWEB_PAY_DOMAIN } from "../../../../constants/urls";
+
+export type PayTopCustomersData = {
+  count: number;
+  customers: Array<{
+    walletAddress: string;
+    totalSpendUSDCents: number;
+  }>;
+};
+
+type Response = {
+  result: {
+    data: PayTopCustomersData;
+  };
+};
+
+export function usePayCustomers(options: {
+  clientId: string;
+  from: Date;
+  to: Date;
+  pageSize: number;
+  type: "top-customers" | "new-customers";
+}) {
+  const { user, isLoggedIn } = useLoggedInUser();
+
+  return useInfiniteQuery({
+    queryKey: ["usePayTopCustomers", user?.address, options],
```

Comment:
nit: this will work, _however_ it is odd to lead the query key with `Top` when it can be either `top` or `new`

only reasons it works is because the type sits in `options`, can probably clean this up

Reply:
yeah - this used to be usePayTopCustomers - forgot to change the key